src/coreclr/jit/lsrabuild.cpp

```diff
@@ -2756,6 +2753,16 @@ void           LinearScan::buildIntervals()
         availableRegCount = REG_INT_COUNT;
     }
 
+    if (availableRegCount < (sizeof(regMaskTP) * 8))
+    {
+        // Mask out the bits that are between 64 ~ availableRegCount
+        actualRegistersMask = (1ULL << availableRegCount) - 1;
+    }
+    else
+    {
+        actualRegistersMask = ~RBM_NONE;
+    }
```

Comment:
Why not have this always be `actualRegisterMask = (1ULL << availableRegCount) - 1`?

That way its always exactly the bitmask of actual registers available. No more, no less.

Reply:
Yes, that's ideally how it should be, but for arm64, `availableRegCount == 65` (including the `REG_STK`, etc.). So `(1ULL << 65)` returns `0x2` and with `- 1`, we get `actualRegisterMask` becomes `1`. Debugger shows correct value.

<img width="506" alt="image" src="https://github.com/dotnet/runtime/assets/12488060/c9ea5831-b286-4710-8eeb-8e78703bdc3e">

I am little confused on why that happens.

Reply:
> Some compilers are going to do overshifting as if we had infinite bits and then truncated. This would make it (1 << 65) == 0, then 0 - 1 == -1, which is AllBitsSet. Other compilers are going to instead treat this as C# and x86/x64 do, which is to treat it as (1 << (65 % 63)) == (1 << 2) ==  4 and then 4 - 1 == 3 and others still as something completely different.

That's exactly what I understand it. What confuses me is the compiler decides different behavior during execution vs. "watch windows" in debugging.

Reply:
While I agree with your suggestion, for this PR, I will keep the code that I have currently to handle the arm64 case.